Update rke-hardening-guide based on CIS-1.24 and CIS-1.7 (#794)
rancher/rancher-docs: docs/pages-for-subheaders/rke1-hardening-guide.md

| Rancher Version | CIS Benchmark Version | Kubernetes Version           |
|-----------------|-----------------------|------------------------------|

> | Rancher v2.7    | Benchmark v1.7       | Kubernetes v1.24 up to v1.25 |

Author: As per @Priyashetty17 suggestion.
```suggestion
| Rancher v2.7    | Benchmark v1.7       | Kubernetes v1.25 |
```

Reviewer: @andypitcher I did not realize this was the hardening guide. I got confused with the self assessment guide for v1.7. Since the upstream only supports cis v1.23 as of now, should we leave the CIS Benchmark Version in the Hardening Guide at 1.23 and K8s version "v1.23 up to v1.26"?

Author: This is a good point, we could have one hardening-guide page for each distro, with one tab page for each CIS version (v1.23, v1.24 and v1.7) then, since cis-1.23 is the only option available upstream (it means we need to develop/provide cis-1.24 and cis-1.7 upstream).
@macedogm do you have some recommendation ?

Reviewer 2: @andypitcher I partially agree with the proposal:

> we could have one hardening-guide page for each distro

We already have this.

> with one tab page for each CIS version (v1.23, v1.24 and v1.7)

True. For the moment each tab is named as the K8s version, we would only have to change it to be named according to the CIS version too, perhaps from:

`v1.25 and Newer | v1.24 and Older`

To:

`CIS 1.7 (K8s v1.25) | CIS 1.24 (K8s v1.24) | CIS 1.23 / (K8s v1.22-v1.23)`

Is that what you mean?

![image](https://github.com/rancher/rancher-docs/assets/17654553/1e49bf13-3e15-4fbf-8d08-27e599881e48)

> then, since cis-1.23 is the only option available upstream (it means we need to develop/provide cis-1.24 and cis-1.7 upstream).

I didn't get this part. Can you please elaborate further?

Reviewer 2: Thanks for the explanation. While the RKE2 flag isn't improved, should we just mention that users must use the `cis-1.23` for all CIS versions then?

> | Rancher v2.7    | Benchmark v1.7       | Kubernetes v1.24 up to v1.25 |

Reviewer 2: Any reason why we are removing the versions < 1.25 from the hardening guide? I didn't get this change.
Rancher v2.7 still supports K8s v1.23 to v1.24, so we need to maintain them there.
Ideally the change should be something like this, unless I'm missing some context:

```suggestion
| Rancher v2.7    | Benchmark v1.23       | Kubernetes v1.23             |
| Rancher v2.7    | Benchmark v1.24       | Kubernetes v1.24             |
| Rancher v2.7    | Benchmark v1.7        | Kubernetes v1.25 up to v1.26 |
```

:::note

> - Since Benchmark v1.24, check id `4.1.7 Ensure that the certificate authorities file permissions are set to 600 or more restrictive (Automated)` might fail, as /etc/kubernetes/ssl/kube-ca.pem is provisioned in 644 by default.
> - Since Benchmark v1.7 (latest), `--protect-kernel-defaults` (check id 4.2.6) parameter is not required anymore, and was replaced.

Reviewer 3: ```suggestion
- In Benchmark v1.24 and later, check id `4.1.7 Ensure that the certificate authorities file permissions are set to 600 or more restrictive (Automated)` might fail, as /etc/kubernetes/ssl/kube-ca.pem is provisioned in 644 by default.
- In Benchmark v1.7, `--protect-kernel-defaults` (check id 4.2.6) parameter is not required anymore, and was replaced.
```

The danger in noting "latest" is that we'll have to remember to update when it's no longer latest.

Author: Suggestion updated.

Reviewer 2: @andypitcher I believe we only need to address this missing feedback from Marty, right?

> The danger in noting "latest" is that we'll have to remember to update when it's no longer latest.

  kubelet:
    extra_args:
      feature-gates: RotateKubeletServerCertificate=true

>       protect-kernel-defaults: "true"

Reviewer 2: This needs to be inside a specific tab depending on the CIS benchmark version, because it's still needed in CIS 1.23 and CIS 1.24.

  kubelet:
    extra_args:
      feature-gates: RotateKubeletServerCertificate=true

>       protect-kernel-defaults: true

Reviewer 2: Same as before:

> This needs to be inside a specific tab depending on the CIS benchmark version, because it's still needed in CIS 1.23 and CIS 1.24.

Reviewer 2: This flag was removed, but it still need to be kept because it's inside the v1.24 tab, no?
